### PR TITLE
WARM-IDE: Add check for invalid project name when creating new project

### PR DESCRIPTION
**Pull request by NingW101**

### CODE REVIEW

File: test-tools/wamr-ide/VSCode-Extension/src/view/NewProjectPanel.ts [hunk 1]
```diff
@@ -178,16 +186,21 @@ export class NewProjectPanel {
...
                             vscode.window.showErrorMessage(
                                 'Please fill chart before your submit!'
                             );
                             return;
+                        } else if (
+                            createNewProjectStatus ===
+                            NewProjectPanel.DIR_PATH_INVALID_ERR
+                        ) {
+                            vscode.window.showErrorMessage(
+                                `A file name can't contain any of the following charachters: / \\ : * ? < > | " and length less than 255`
```
Comment (wenyongh) [inline test-tools/wamr-ide/VSCode-Extension/src/view/NewProjectPanel.ts:201]: Can the string be wrapped by `? Note that it isn't '

Comment (NingW101) [reply test-tools/wamr-ide/VSCode-Extension/src/view/NewProjectPanel.ts]: Updated

File: test-tools/wamr-ide/VSCode-Extension/src/utilities/directoryUtilities.ts [hunk 2]
```diff
@@ -94,3 +95,24 @@ export function CheckIfDirectoryExist(path: string): boolean {
...
+export function checkFolderName(folderName: string) {
+    var status = true;
+    let OS_PLATFORM = os.platform();
+    if (folderName.length > 255) {
+        status = false;
+    }
+    let errArr: string[] = [];
+    if (OS_PLATFORM === 'win32') {
+        errArr = ['\\', '/', ':', '?', '*', '"', '|', '<', '>'];
+    } else if (OS_PLATFORM === 'linux') {
```
Comment (yamt) [inline test-tools/wamr-ide/VSCode-Extension/src/utilities/directoryUtilities.ts:108]: add `|| OS_PLATFORM === 'darwin'`

or, maybe it's simpler to reject any characters which are invalid on any of platforms.

File: test-tools/wamr-ide/VSCode-Extension/src/view/NewProjectPanel.ts [hunk 3]
```diff
@@ -178,16 +186,21 @@ export class NewProjectPanel {
...
                             vscode.window.showErrorMessage(
                                 'Please fill chart before your submit!'
                             );
                             return;
+                        } else if (
+                            createNewProjectStatus ===
+                            NewProjectPanel.DIR_PATH_INVALID_ERR
+                        ) {
+                            vscode.window.showErrorMessage(
+                                "A file name can't contain any of the following characters: ' / \\ : * ? < > | ' and the length should be less than 255"
```
Comment (yamt) [inline test-tools/wamr-ide/VSCode-Extension/src/view/NewProjectPanel.ts:201]: does this message assume windows?

File: test-tools/wamr-ide/VSCode-Extension/src/utilities/directoryUtilities.ts (hunk 2 above)
Comment (NingW101) [reply test-tools/wamr-ide/VSCode-Extension/src/utilities/directoryUtilities.ts]: Thanks for the comment, updated

File: test-tools/wamr-ide/VSCode-Extension/src/view/NewProjectPanel.ts (hunk 3 above)
Comment (NingW101) [reply test-tools/wamr-ide/VSCode-Extension/src/view/NewProjectPanel.ts]: Updated. Thanks for your review

**Pull request merged**